test/unitary/IQueryable.test.ts

```diff
@@ -1045,6 +1046,116 @@ export namespace IQueryableUnitTest
         });
     }
 
+    function sequenceEqual(instancer: Instancer): void
+    {
+        class Person
+        {
+            constructor(public firstName: string, public lastName: string, public age: number) {}
+        }
+
+        it("Both empty return true", () => 
+        {
+            const first = instancer([]);
+            const second = instancer([]);
+            
+            Test.isTrue(first.sequenceEqual(second));
+        });
+
+        it("Different count return false (number)", () => 
+        {
+            const first = instancer<number>([]);
+            const second = instancer<number>([1]);
+            
+            Test.isFalse(first.sequenceEqual(second));
+        });
+
+        it("Same count different values return false (number)", () => 
+        {
+            const first = instancer<number>([2]);
+            const second = instancer<number>([1]);
+            
+            Test.isFalse(first.sequenceEqual(second));
+        });
+
+        it("Different count return false (string)", () => 
+        {
+            const first = instancer<string>([]);
+            const second = instancer<string>(["test1"]);
+            
+            Test.isFalse(first.sequenceEqual(second));
+        });
+
+        it("Same count different values return false (string)", () => 
+        {
+            const first = instancer<string>(["test1"]);
+            const second = instancer<string>(["test2"]);
+            
+            Test.isFalse(first.sequenceEqual(second));
+        });
+
+        it("Same object in both return true", () => 
+        {
+            const obj = {};
+
+            const first = instancer<any>([obj]);
+            const second = instancer<any>([obj]);
+            
+            Test.isTrue(first.sequenceEqual(second));
+        });
+
+        const stringLengthComparer = (left: string, right: string) => left.length === right.length;
+
+        it("Custom comparer; check lengths; should return true (string)", () => 
+        {
+            const first = instancer<string>(["one"]);
+            const second = instancer<string>(["two"]);
+            
+            Test.isTrue(first.sequenceEqual(second, stringLengthComparer));
+        });
+
+        it("Custom comparer; check lengths; should return false (string)", () => 
+        {
+            const first = instancer<string>(["three"]);
+            const second = instancer<string>(["four"]);
+            
+            Test.isFalse(first.sequenceEqual(second, stringLengthComparer));
+        });
+
+        const personAgeAndFirstNameComparer = (left: Person, right: Person) => left.age === right.age && left.firstName === right.firstName;
```

Comment:
```ts
const personAgeAndFirstNameComparer = (left: Person, right: Person)
    => left.age === right.age && left.firstName === right.firstName;
```

Reply:
Line terminator is not permitted before arrow, i can write like this if you want:

    const personAgeAndFirstNameComparer = (left: Person, right: Person) =>
        left.age === right.age && left.firstName === right.firstName;

Reply:
yep, just for readability on laptops 😋